### PR TITLE
add ability to specify healthcheck protocol

### PR DESCRIPTION
**Pull request by ssalinas**
enables us to do healthchecks on services only listening for ssl traffic

@tpetr

### CODE REVIEW

File: SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityHealthchecker.java [hunk 1]
```diff
@@ -148,7 +150,17 @@ public void reEnqueueOrAbort(SingularityTask task) {
...
 
-    return Optional.of(String.format("http://%s:%d/%s", hostname, firstPort.get(), uri));
+    String scheme = getHealthcheckScheme(task.getTaskRequest().getDeploy());
+
+    return Optional.of(String.format("%s://%s:%d/%s", scheme, hostname, firstPort.get(), uri));
+  }
+
+  private String getHealthcheckScheme(SingularityDeploy deploy) {
+    if (deploy.getHealthcheckScheme().isPresent()) {
+      return deploy.getHealthcheckScheme().get();
```
Comment (tpetr) [inline SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityHealthchecker.java]: i'd just use `.or(DEFAULT_HEALTH_CHECK_SCHEME)`

File: SingularityBase/src/main/java/com/hubspot/singularity/SingularityDeploy.java [hunk 2]
```diff
@@ -81,7 +82,8 @@ public SingularityDeploy(@JsonProperty("requestId") String requestId,
       @JsonProperty("loadBalancerGroups") Optional<List<String>> loadBalancerGroups,
       @JsonProperty("considerHealthyAfterRunningForSeconds") Optional<Long> considerHealthyAfterRunningForSeconds,
       @JsonProperty("loadBalancerOptions") Optional<Map<String, Object>> loadBalancerOptions,
-      @JsonProperty("skipHealthchecksOnDeploy") Optional<Boolean> skipHealthchecksOnDeploy) {
+      @JsonProperty("skipHealthchecksOnDeploy") Optional<Boolean> skipHealthchecksOnDeploy,
+      @JsonProperty("healthCheckScheme") Optional<String> healthcheckScheme) {
```
Comment (tpetr) [inline SingularityBase/src/main/java/com/hubspot/singularity/SingularityDeploy.java]: hate to nitpick but i think `healthcheckProtocol` is more accurate. also considering the fact that we only support http(s), might want to make it an enum instead

**Comment by ssalinas** [comment]
@tpetr fixed to use an enum and name to healthcheckProtocol

File: SingularityBase/src/main/java/com/hubspot/singularity/HealthcheckProtocol.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.hubspot.singularity;
+
+public enum HealthcheckProtocol {
+
+  http, https;
```
Comment (tpetr) [inline SingularityBase/src/main/java/com/hubspot/singularity/HealthcheckProtocol.java]: enum entries should be in all caps

File: SingularityBase/src/main/java/com/hubspot/singularity/SingularityDeploy.java [hunk 4]
```diff
@@ -81,7 +82,8 @@ public SingularityDeploy(@JsonProperty("requestId") String requestId,
       @JsonProperty("loadBalancerGroups") Optional<List<String>> loadBalancerGroups,
       @JsonProperty("considerHealthyAfterRunningForSeconds") Optional<Long> considerHealthyAfterRunningForSeconds,
       @JsonProperty("loadBalancerOptions") Optional<Map<String, Object>> loadBalancerOptions,
-      @JsonProperty("skipHealthchecksOnDeploy") Optional<Boolean> skipHealthchecksOnDeploy) {
+      @JsonProperty("skipHealthchecksOnDeploy") Optional<Boolean> skipHealthchecksOnDeploy,
+      @JsonProperty("healthCheckScheme") Optional<HealthcheckProtocol> healthcheckProtocol) {
```
Comment (tpetr) [inline SingularityBase/src/main/java/com/hubspot/singularity/SingularityDeploy.java]: don't forget to update the annotation, too

File: SingularityBase/src/main/java/com/hubspot/singularity/SingularityDeploy.java [hunk 5]
```diff
@@ -234,6 +238,11 @@ public String getId() {
     return healthcheckUri;
   }
 
+  @ApiModelProperty(required=false, value="Healthcheck scheme")
```
Comment (tpetr) [inline SingularityBase/src/main/java/com/hubspot/singularity/SingularityDeploy.java]: update description

**Comment by ssalinas** [comment]
updated

**Comment by tpetr** [comment]
LGTM, thanks

**Pull request merged**